tests/unit/test_xapidb_filter.py

```diff
@@ -55,9 +54,22 @@
 """
 
 
+def assert_xml_element_trees_equiv(a, b):
+    """Assert that the contents of two XML ElementTrees are equivalent (recursive)"""
+    assert a.tag == b.tag
+    assert (a.text or "").strip() == (b.text or "").strip()
+    assert (a.tail or "").strip() == (b.tail or "").strip()
+    assert a.attrib.items() == b.attrib.items()
+    for achild, bchild in zip(list(a), list(b)):
```

Comment:
What happens when the two lists have different length?

Reply:
@lindig Thank you, good catch and well-spoken! Was ignored!

Fixed and now I'm keeping the check using `xml.dom.minidom.parseString()` as it can still be used for Python2 or Python3 to double-check!